### PR TITLE
fix(distro): Remove deprecated distros from the code

### PR DESCRIPTION
**Pull request by fruch**
remove the following distros from code:
- ubuntu 16
- ubuntu 18
- debian 8
- debian 9

those distro doesn't have any job using them anymore,
and now we remove all code that is referring to them

##### artifacts 
- [x] https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/artifacts-ubuntu2204-test/37/
- [x] https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/artifacts-ubuntu2004-arm-test/7/
- [x] https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/artifacts-centos8-test/12/
- [x] https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/artifacts-rocky9-test/27/
- [x] https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/artifacts-offline-install_artifacts-ubuntu2004-test/5/
##### upgrade
- [x] https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/rolling-upgrade-centos8-test/11/
- [x] https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/rolling-upgrade-debian11-test/6/
- [x] https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/rolling-upgrade-ubuntu20.04-test/15/
##### longevity 
- [x] - one aws longevity - https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-100gb-4h-test/26/
- [x] - one gcp longevity - https://jenkins.scylladb.com/job/scylla-staging/job/fruch/job/longevity-10gb-3h-gce-test/40/
- [x] - provision tests

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1870,55 +1870,27 @@ def install_scylla(self, scylla_repo, scylla_version=None):
...
-                    apt-get install software-properties-common -y
-                    apt-get install openjdk-11-jre -y
-                """)
-                self.remoter.run('sudo bash -cxe "%s"' % install_debian_10_prereqs)
-
-            self.remoter.run(
-                'sudo DEBIAN_FRONTEND=noninteractive apt-get '
-                '-o Dpkg::Options::="--force-confold" '
-                '-o Dpkg::Options::="--force-confdef" '
-                'upgrade -y ')
```
[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster.py:2035]: first, it's not part of the installation instructions of scylla, and we not doing it for yum base installs for example.
second, it's not needed when we run artifacts on GCE, we start from latest available image.

there a seperate function that does this, and it used in rolling upgrade, where is part of the documented upgrade procedure

I followed it back, and it seems to be it was introduced without clear reasoning and consideration.

I will gonna test those regardless, but there's no real reason to do that on each install

[the remark this answers is not in the document]
Comment (fruch) [reply sdcm/cluster.py:2035]: so for all of the install passed, with this part.

**Comment by fruch** [comment]
so far most of the provision tests passed, all the artifacts pass, and now waiting for upgrade test and longevities

**Comment by roydahan** [comment]
I'm considering giving prizes for submitters who send PRs that remove more code lines than adding.

**Comment by fruch** [comment]
> I'm considering giving prizes for submitters who send PRs that remove more code lines than adding.

maybe every sprints, only for dropping code.
and you know where to send the monster coin...

**Review by vponomaryov — APPROVED** [review]
LGTM

**Review by soyacz — APPROVED** [review]
LGTM

**Pull request merged**